### DISCUSSION TITLE
Name and URL unintentionally changed via Public API

### DISCUSSION THREAD

**Discussion by ecxs** [Budibase/budibase#9148] [Help] [answered]
I was playing around with the Public API and all of a sudden the Name and Url Path of the App in the top of my list changed from Asset Management to est eu ullamco !? Now I want to change this back, but for some reason or another, the Edit button is greyed out. Please help!!!

![Screenshot 2022-12-22 at 09 57 09](https://user-images.githubusercontent.com/4005405/209097689-107dd1e2-64da-45ea-bac0-433d9009cc5f.png)

![Screenshot 2022-12-22 at 09 56 02](https://user-images.githubusercontent.com/4005405/209097717-14cb31a6-e7af-493b-8805-b1d3522674b0.png)

![Screenshot 2022-12-22 at 09 58 36](https://user-images.githubusercontent.com/4005405/209097736-8bb484f0-2286-442e-a551-809c145ecf80.png)

**Comment by shogunpurple** [comment, marked as answer]
Hi @ecxs - be careful! The API is quite powerful and allows you to update almost any data, so mistakes can be costly.

You could just use the update endpoint to reset your app name back to what it was.

The reason your edit button is grayed out is because your app is published. If you unpublish your app, you can change it back then republish.

**Reply by ecxs** [reply]
Hi @shogunpurple thank! I just unpublished the app and the name and Url Path reverted back automagically ;-)

You probably want to check your Postman collection, as the 'Retrieve an application' request suggests that is is a GET, but actually is a put ...

**Reply by shogunpurple** [reply]
@ecxs are you sure? That endpoint is working at a GET for me and we have automated tests for the public API. Is it possible you used a POST or PUT on that endpoint instead of a GET?

**Reply by ecxs** [reply]
@shogunpurple I'm not usually changing imported Postman collections, but of course I might have done this accidentally. But this is wat I've got in my Postman collection that resulted in changing the App's Name and Url Path ...
![Screenshot 2022-12-22 at 12 47 23](https://user-images.githubusercontent.com/4005405/209128367-31d4204e-ec9a-47df-9fdf-c6be29797917.png)
